### PR TITLE
[cart-item] Add an action to the front insurance controller to add the insurance product to the basket

### PR DESCRIPTION
**Pull request by Benjamin-Freoua-Alma**
### Reason for change

[Linear task](https://linear.app/almapay/issue/ECOM-1596/[🧩-ps]-add-an-action-to-the-front-insurance-controller-to-add-the)

### How to test

_As a reviewer, you are encouraged to test the PR locally._

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] You understand the impact of this PR on existing code/features
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

File: alma/controllers/hook/FrontHeaderHookController.php [hunk 1]
```diff
@@ -316,6 +316,8 @@ public function manageInsuranceAssetsAfter17()
                 ConstantsHelper::INSURANCE_PRODUCT_CSS_ID,
                 "modules/$this->moduleName/" . ConstantsHelper::INSURANCE_PRODUCT_CSS_PATH
             );
+
+            return '<script type="module" src="https://protect.sandbox.almapay.com/displayModal.js"></script><div id="alma-insurance-modal"></div>';
```
Comment (ClaireAlma) [inline alma/controllers/hook/FrontHeaderHookController.php:320]: une constante serait mieux pr stocket l'url

Comment (Benjamin-Freoua-Alma) [reply alma/controllers/hook/FrontHeaderHookController.php:320]: @ClaireAlma 
Yes, It was for the explo. It's why the PR is still in Draft :)

File: alma/controllers/front/insurance.php [hunk 2]
```diff
@@ -219,4 +234,14 @@ protected function decreaseCartFromOneProduct($context, $idProduct, $idProductAt
             $context->cart->id_address_delivery
         );
     }
+
+    protected function addInsuranceProductAndAssociation()
```
Comment (ClaireAlma) [inline alma/controllers/front/insurance.php:238]: missing doc comment

File: alma/lib/Helpers/InsuranceHelper.php [hunk 3]
```diff
@@ -88,9 +100,18 @@ public function __construct(
     public function isInsuranceAllowedInProductPage()
     {
         return (bool) version_compare(_PS_VERSION_, '1.7', '>=')
```
Comment (ClaireAlma) [inline alma/lib/Helpers/InsuranceHelper.php:102]: use new version helper function

File: alma/lib/Helpers/InsuranceHelper.php [hunk 4]
```diff
@@ -88,9 +100,18 @@ public function __construct(
...
+            && (bool) (int) $this->settingsHelper->getKey(ConstantsHelper::ALMA_SHOW_INSURANCE_WIDGET_PRODUCT, false)
+            && $this->isInsuranceActivated();
+    }
+
+    /**
+     * @return bool
+     */
+    public function isInsuranceAllowedInCartPage()
+    {
+        return (bool) version_compare(_PS_VERSION_, '1.7', '>=')
```
Comment (ClaireAlma) [inline alma/lib/Helpers/InsuranceHelper.php:112]: use new version helper function

File: alma/lib/Helpers/InsuranceHelper.php [hunk 5]
```diff
@@ -99,8 +120,8 @@ public function isInsuranceAllowedInProductPage()
     public function isInsuranceActivated()
     {
         return (bool) version_compare(_PS_VERSION_, '1.7', '>=')
```
Comment (ClaireAlma) [inline alma/lib/Helpers/InsuranceHelper.php:122]: use new version helper function

**Review by ClaireAlma — CHANGES_REQUESTED** [review]

**Review by ClaireAlma — APPROVED** [review]
Review but not tested

**Pull request merged**